### PR TITLE
Update tutorial of memory view

### PR DESCRIPTION
**Pull request by cloudhan**
The memory profiling will be improved in python 1.10. This PR update the corresponding tutorial.

### CODE REVIEW

File: intermediate_source/tensorboard_profiler_tutorial.py [hunk 1]
```diff
@@ -287,28 +289,46 @@ def train(data):
...
 # .. image:: ../../_static/img/profiler_memory_view.png
 #    :scale: 25 %
 #
+# The components are memory curve graph, memory events table and memory statistics table, from top to bottom, respectively.
+#
 # The memory type could be selected in "Device" selection box.
-# For example, "GPU0" means the following table only shows each operator’s memory usage on GPU 0, not including CPU or other GPUs.
+# For example, "GPU0" means the following table only shows each operator's memory usage on GPU 0, not including CPU or other GPUs.
+#
+# The memory curve shows the trends of memory consumption. You can left click and drag on it to select events in the
```
Comment (gaoteng-git) [inline intermediate_source/tensorboard_profiler_tutorial.py:312]: Add explanation of "Allocated" and "Reserved"?

File: intermediate_source/tensorboard_profiler_tutorial.py [hunk 2]
```diff
@@ -287,28 +289,46 @@ def train(data):
...
+#    :scale: 25 %
 #
-# The "Size Increase" sums up all allocation bytes and minus all the memory release bytes.
+# In the memory events table, the allocation and release events are paired into one entry. The "operator" column shows
+# the immediate ATen operator that is causing the allocation.  The "Allocation Time", "Release Time" and "Duration"
+# columns' data might be missing if the event occurs outside of the time range.
 #
-# The "Allocation Size" sums up all allocation bytes without considering the memory release.
+# In the memory statistics table, the "Size Increase" column sums up all allocation size and minus all the memory
+# release size, that is, the net increase of memory usage after this opeartor. And the "Allocation Size" column sums up
```
Comment (gaoteng-git) [inline intermediate_source/tensorboard_profiler_tutorial.py:330]: operator

File: intermediate_source/tensorboard_profiler_tutorial.py [hunk 3]
```diff
@@ -287,28 +289,46 @@ def train(data):
...
+# For example, "GPU0" means the following table only shows each operator's memory usage on GPU 0, not including CPU or other GPUs.
+#
+# The memory curve shows the trends of memory consumption. You can left click and drag on it to select events in the
+# desired range:
+#
+# .. image:: ../../_static/img/profiler_memory_curve_selecting.png
+#    :scale: 25 %
+#
+# After selection, the three components will be updated for the restricted time range, so that you can gain more
+# infomation about it. By repeating this process, you can zoom into very fine-grained detail. Right click on the graph
```
Comment (gaoteng-git) [inline intermediate_source/tensorboard_profiler_tutorial.py:319]: information

File: intermediate_source/tensorboard_profiler_tutorial.py [hunk 4]
```diff
@@ -287,28 +289,46 @@ def train(data):
...
 #
-# The "Size Increase" sums up all allocation bytes and minus all the memory release bytes.
+# In the memory events table, the allocation and release events are paired into one entry. The "operator" column shows
+# the immediate ATen operator that is causing the allocation.  The "Allocation Time", "Release Time" and "Duration"
+# columns' data might be missing if the event occurs outside of the time range.
 #
-# The "Allocation Size" sums up all allocation bytes without considering the memory release.
+# In the memory statistics table, the "Size Increase" column sums up all allocation size and minus all the memory
+# release size, that is, the net increase of memory usage after this opeartor. And the "Allocation Size" column sums up
+# all allocation without considering the memory release, instead.
```
Comment (gaoteng-git) [inline intermediate_source/tensorboard_profiler_tutorial.py:331]: Is it better to explain difference of "Self Size Increase" with "Size Increase"? User may be confused that there are 5 operators with same "Size Increase" value.

Comment (cloudhan) [reply intermediate_source/tensorboard_profiler_tutorial.py]: I added some low level details to help the user to obtain a better understanding.

File: intermediate_source/tensorboard_profiler_tutorial.py (hunk 1 above)
Comment (cloudhan) [reply intermediate_source/tensorboard_profiler_tutorial.py]: done

File: intermediate_source/tensorboard_profiler_tutorial.py [hunk 5]
```diff
@@ -287,28 +289,51 @@ def train(data):
...
 #
-# The "Size Increase" sums up all allocation bytes and minus all the memory release bytes.
+# In the memory events table, the allocation and release events are paired into one entry. The "operator" column shows
+# the immediate ATen operator that is causing the allocation.  The "Allocation Time", "Release Time" and "Duration"
+# columns' data might be missing if the event occurs outside of the time range.
 #
-# The "Allocation Size" sums up all allocation bytes without considering the memory release.
+# In the memory statistics table, the "Size Increase" column sums up all allocation size and minus all the memory
+# release size, that is, the net increase of memory usage after this operator. The "Self Size Increase" column is
+# similar to "Size Increase", but it does not count children operators allocation. With regards to ATen operators'
```
Comment (gaoteng-git) [inline intermediate_source/tensorboard_profiler_tutorial.py:333]: operators' allocation

**Review by gaoteng-git — APPROVED** [review]

**Comment by cloudhan** [comment]
@chauhang Please have a look

**Review by gaoteng-git — APPROVED** [review]
LGTM

**Pull request merged**